frontend/packages/operator-lifecycle-manager/src/utils.ts

```diff
@@ -0,0 +1,32 @@
+import * as _ from 'lodash';
+import { Annotation, ClusterServiceVersionKind } from './types';
+import { referenceForProvidedAPI } from './components';
+
+export const getInternalObjects = (annotations: Annotation): string[] => {
+  const internals: string = _.get(annotations, 'operators.operatorframework.io/internal-objects');
+  if (internals) {
+    try {
+      return JSON.parse(internals);
+    } catch {
+      return [];
+    }
+  }
+  return [];
+};
+
+export const isInternalObject = (internalObjects: string[], objectName: string): boolean =>
+  internalObjects.some((obj) => obj === objectName);
+
+export const internalAPIReferences = (
+  internalObjects: string[],
```

Comment:
I'd just parse the internal objects here instead of taking it in as a param. That makes the API simpler.